api/deployment/DeploymentConfigRestHandler.go

```diff
@@ -0,0 +1,124 @@
+package deployment
+
+import (
+	"github.com/devtron-labs/devtron/api/restHandler/common"
+	chartRepoRepository "github.com/devtron-labs/devtron/pkg/chartRepo/repository"
+	"github.com/devtron-labs/devtron/pkg/pipeline"
+	"github.com/devtron-labs/devtron/pkg/sql"
+	"github.com/devtron-labs/devtron/pkg/user"
+	"github.com/devtron-labs/devtron/pkg/user/casbin"
+	"github.com/juju/errors"
+	"go.uber.org/zap"
+	"gopkg.in/go-playground/validator.v9"
+	"io/ioutil"
+	"net/http"
+	"os"
+	"time"
+)
+
+type DeploymentRestHandler interface {
```

Comment:
can we change interface and impl object from Deployment to DeploymentConfig, so that one does not gets it confused with deployment operation